docs/source/lightning_cli.rst

```diff
@@ -0,0 +1,341 @@
+      --trainer.min_epochs MIN_EPOCHS
+                            Force training for at least these many epochs (type: int,
+                            default: 1)
+      ...
+
+    Example encoder-decoder model:
+      --model.encoder_layers ENCODER_LAYERS
+                            Number of layers for the encoder (type: int, default: 12)
+      --model.decoder_layers DECODER_LAYERS
+                            Number of layers for each decoder block (type: List[int],
+                            default: [2, 4])
+
+The default configuration that option :code:`--print_config` gives is in yaml
+format and for the example above would look as follows:
+
+.. code-block:: bash
+
+    $ python trainer.py --print_config
+    model:
+      decoder_layers:
+      - 2
+      - 4
+      encoder_layers: 12
+    trainer:
+      accelerator: null
+      accumulate_grad_batches: 1
+      amp_backend: native
+      amp_level: O2
+      ...
+
+Note that there is a section for each class (model and trainer) including all
+the init parameters of the class. This grouping is also used in the formatting
+of the help shown previously.
+
+
+Trainer Callbacks and arguments with class type
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A very important argument of the
+:class:`~pytorch_lightning.trainer.trainer.Trainer` class is the
+:code:`callbacks`. In contrast to other more simple arguments which just require
+numbers or strings, :code:`callbacks` expects a list of instances of subclasses
+of :class:`~pytorch_lightning.callbacks.Callback`. To specify this kind of
+argument in a config file, each callback must be given as a dictionary including
+a :code:`class_path` entry with an import path of the class, and optionally an
+:code:`init_args` entry with arguments required to instantiate it. Therefore, a
+simple configuration file example that defines a couple of callbacks is the
+following:
+
+.. code-block:: yaml
+
+    trainer:
+      callbacks:
+        - class_path: pytorch_lightning.callbacks.EarlyStopping
+          init_args:
+            patience: 5
+        - class_path: pytorch_lightning.callbacks.LearningRateMonitor
+          init_args:
+            ...
+
+Similar to the callbacks, any arguments in
+:class:`~pytorch_lightning.trainer.trainer.Trainer` and user extended
+:class:`~pytorch_lightning.core.lightning.LightningModule` and
+:class:`~pytorch_lightning.core.datamodule.LightningDataModule` classes that
+have as type hint a class can be configured the same way using
+:code:`class_path` and :code:`init_args`.
+
+
+Multiple models and/or datasets
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In the previous examples :class:`~pytorch_lightning.utilities.cli.LightningCLI`
+works only for a single model and datamodule class. However, there are many
+cases in which the objective is to easily be able to run many experiments for
+multiple models and datasets. For these cases the tool can be configured such
```

Comment:
> However, there are many cases in which the objective is to easily be able to run many experiments for multiple models and datasets.

I fully agree with that. 

However, the current solution requires the user to create a shared config for each model/data/trainer combination, which doesn't scale. 

Could we have a config file for each part? So that all combinations can be made from the command line? 

So that something like this works out of the box
```bash
python trainer.py --model modelA.yml --data dataA.yml 
python trainer.py --model modelA.yml --data dataB.yml 
python trainer.py --model modelB.yml --data dataA.yml 
python trainer.py --model modelB.yml --data dataB.yml 
```

Reply:
What you suggest should already be working. I haven't tested it explicitly with LightningCLI, but that feature is enabled. The relevant part of the code is https://github.com/omni-us/jsonargparse/blob/master/jsonargparse/signatures.py#L301